### ISSUE TITLE
Open project in different type of printer

### ISSUE THREAD

**Issue by BagelOrb** [Ultimaker/Cura#10496]
### Is your feature request related to a problem?

I want to share my designs with with the recommended setting adjustments in the project file, but I would like those project files to be openable by any type of printer.

### Describe the solution you'd like

Have an extra button in this dropdown called [open in current printer]
![image](https://user-images.githubusercontent.com/8895761/134336682-b5921a61-16b9-4e49-ab24-3a0094eed48a.png)

The setting overrides from the 3mf file are then loaded as setting changes on the currently selected printer and current profile.

Alternatively, the dropdown above contains all listed printers.

Optionally, the user is asked which base profile to put the overrides on.

### Describe alternatives you've considered

Being able to save user profiles as a separate thing from quality profiles. This would require a total rework of the settings system, so it's not a good idea.

### Affected users and/or printers

Mostly in situations where the designer is a 3D printing expert and needs to tell the printer engineer how to print the thing.

Or situations where you want to print a file within the same company, but the one location has an UM3 and the other has an UM S5.

### Additional information & file uploads

Quite often you will find designs on thingiverse with setting recommendations written out. For noob users it can be quite challenging to implement these setting changes, because they don't know how to find advanced settings etc.

Presumably this problem also translates into the professional segment, but it is a lot harder to gather data on that.

**Comment by Ghostkeeper** [comment]
Doesn't this kind of defeat the intent of project files as a way to reproduce a print?

My main concern here is that project files are complicated to the user as it is. It'll override existing settings they might have wanted to save, change settings that they don't even know the existence of, and later on they'll encounter something weird caused by this setting change and not know of the setting to change to fix it because they didn't manually change that setting in the first place. I think this feature would make that problem worse, by expanding the scope of project files to include also cross-contamination of settings from one printer type to another.

As an example of the problem that this feature would address, most models on Thingiverse include instructions to print the object with certain settings (e.g. [this one](https://www.thingiverse.com/thing:4966110) indicating to not use rafts or support and use 15% infill). The very valid point of this feature request is to make that sort of thing easier. But I suppose you know how this will be used: Changing printing temperatures, layer heights, retraction distance and speed, etc.
A good use of this feature would be to override enabling of support when the model needs it, or mesh fixing things like Extensive Stitching. However even in that mesh fixing segment, a certain print may need to enable "Remove All Holes" and when the user prints something else afterwards they'll wonder why certain parts get filled up and be unable to continue their work because of what they think are bugs.
A bad use of this feature would be to override things that are very material or printer dependent but the user might think is model-dependent, such as Maximum Resolution. My aforementioned example from Thingiverse (happened to be the first result on the homepage atm, October looming) already has this problem a bit by saying no to rafts, while the material type might need a raft.

Now, this is already a problem with project files by allowing to update existing printers. This needs to be addressed too, but this feature would make that problem much worse.

**Comment by BagelOrb** [comment]
I don't think it defeats the original intent of project files, as that original intent would still be default behavior.
To me a project file is the file in which I store the project I am working on, which includes the 3D models and their settings, but the printer I will use is less specific to the project I am working on. That contrasts a bit with the original intent of project files, which was to save the whole current state of Cura. However, when I am working on a project I often don't really care with which specific printer it is printed; most of my setting changes come from considerations about the geometry and how it relates to printability of FDM in general. I think it therefore makes sense for a project file to reflect that and also be openable in other printers.

The problems you address are UX problems and they should be addressed separately. I don't think the problems are exacerbated by my proposal - they are just as much a problem in the original scenario as in the new scenario. These problems therefore have no bearing on the question of accepting the requested feature or not. A user can now become confused if setting changes suddenly appear on their printer and exactly the same would hold if they would open a project on a different printer.

I think the solution direction is in how we deal with setting changes when Cura is closed. I think it's more logical if Cura would ask to save the setting changes when closing Cura and always start Cura with no setting changes. That would align well with how most other programs work: you start with a clean document or with a template. We then also force people to reflect on setting changes just after they have introduced them, so that they are not left wondering why their prints suddenly fail the next time they start a print and forgot about those changes.
Alternatively, Cura could shut down without any prompt and at startup ask whether to "recover unsaved documents" as it were.
Also we might want to introduce a preference for legacy behavior for the people who don't like to see any popups when starting/quitting Cura.

However, I think that that's a separate issue not related to this feature request.

As for opening project files on different printers: we could make it more explicit by calling the option "Migrate to printer [x]".

It has happened several times to me that I am working on a project and have to reproduce the whole thing when my S5 is occupied and I want to switch to my UM3 instead.
Settings I often change are 100% top/bottom, make overhang printable, overhang angle, custom support settings, wall line count. infill pattern, fuzzy skin, initial bottom layers, etc.

It should be noted that the per-object settings are migrated when you switch printer. If the GUI for per-object settings would have been better then I would always record such project-related settings in the objects themselves. Currently using the setting sidebar is still the default for employing project related setting changes.
